currency/rate_engines.go

```diff
@@ -0,0 +1,42 @@
+package currency
+
+import "strings"
+
+const (
+	CustomRates = iota
+	PBSRates
+)
+
+type RateEngines struct {
```

Comment:
Love the concept, not the name. The term 'Engine' is meaningless in this context. Consider names along the lines of `AggregateConversions` or `requestRates`.

Reply:
Do you like `ConversionsGroup`? How about `GroupedConversions` or `GroupedCurrencyConversions`? I'm not great at naming.

Reply:
`GroupedCurrencyConversions` too wordy IMHO. If you don't like `AggregateConversions`, then `GroupedConversions` is similar or perhaps also `RequestConversions`?